pkg/kapis/devops/v1alpha3/pipelinerun/handler.go

```diff
@@ -138,7 +138,7 @@ func (h *apiHandler) createPipelineRun(request *restful.Request, response *restf
 	}
 	// create PipelineRun
 	pr := CreatePipelineRun(&pipeline, &payload, scm)
-	if user != nil && user.GetName() != "" {
```

Comment:
Thanks for improving this line. I'm thinking if it's possible to add a unit test for this case. I know that line 133 already makes sure the variable `user` is not nil. But in case other contributors forget that a unit test might be the perfect solution.

This is only a recommendation. We can merge this PR without new unit tests. But please see also the following examples if you want to add new unit tests.

https://github.com/kubesphere/ks-devops/blob/master/pkg/kapis/devops/v1alpha3/steptemplate/route_test.go